clientlibrary/worker/shard-consumer.go

```diff
@@ -30,93 +30,36 @@ package worker
 import (
 	"errors"
 	"math"
-	"sync"
 	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/kinesis"
-	"github.com/aws/aws-sdk-go/service/kinesis/kinesisiface"
-	deagg "github.com/awslabs/kinesis-aggregation/go/deaggregator"
 
 	chk "github.com/vmware/vmware-go-kcl/clientlibrary/checkpoint"
-	"github.com/vmware/vmware-go-kcl/clientlibrary/config"
 	kcl "github.com/vmware/vmware-go-kcl/clientlibrary/interfaces"
 	"github.com/vmware/vmware-go-kcl/clientlibrary/metrics"
-	par "github.com/vmware/vmware-go-kcl/clientlibrary/partition"
 )
 
-const (
-	// This is the initial state of a shard consumer. This causes the consumer to remain blocked until the all
-	// parent shards have been completed.
-	WaitingOnParentShards ShardConsumerState = iota + 1
-
-	// ErrCodeKMSThrottlingException is defined in the API Reference https://docs.aws.amazon.com/sdk-for-go/api/service/kinesis/#Kinesis.GetRecords
-	// But it's not a constant?
-	ErrCodeKMSThrottlingException = "KMSThrottlingException"
-)
-
-type ShardConsumerState int
-
 // ShardConsumer is responsible for consuming data records of a (specified) shard.
 // Note: ShardConsumer only deal with one shard.
 type ShardConsumer struct {
```

Comment:
We could rename the file to polling-shard-consumer.go to be consistent with AWS java KCL.
https://github.com/awslabs/amazon-kinesis-client/tree/cce05bf4fd03eed007bf6c51203f3cccc189c92a/amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling